crates/neon/src/types_impl/map.rs

```diff
@@ -0,0 +1,163 @@
+use std::{error, fmt, mem::MaybeUninit};
+
+use crate::{
+    context::{internal::Env, Context},
+    handle::{internal::TransparentNoCopyWrapper, root::NapiRef, Handle},
+    macro_internal::NeonResultTag,
+    object::Object,
+    result::{JsResult, NeonResult, ResultExt},
+    sys::{self, raw},
+    types::{private, JsFunction, JsObject, Value},
+};
+
+use super::{JsBoolean, JsNumber, JsUndefined};
+
+#[derive(Debug)]
+#[repr(transparent)]
+/// The type of JavaScript
+/// [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)
+/// objects.
+pub struct JsMap(raw::Local);
+
+impl JsMap {
+    pub fn new<'cx, C>(cx: &mut C) -> NeonResult<Handle<'cx, Self>>
+    where
+        C: Context<'cx>,
+    {
+        let map = cx
+            .global::<JsFunction>("Map")?
+            .construct_with(cx)
+            .apply::<JsObject, _>(cx)?;
+
+        Ok(map.downcast_or_throw(cx)?)
+    }
+
+    pub fn size<'a, C: Context<'a>>(&self, cx: &mut C) -> NeonResult<Handle<'a, JsNumber>> {
```

Comment:
Similar to other methods, I think it's more valuable to immediately extract the value. The user will almost always need the `f64` if they are calling this and it's more ergonomic in addition to being one fewer Node-API call.

```rust
impl JsMap {
    pub fn size<'cx>(&self, cx: &mut Cx<'cx>) -> NeonResult<f64> {
        self.prop("size").get()
    }
}
```